### ISSUE TITLE
[Error: ENOENT: no such file or directory, open '7476980-b2dda329-72ea-40b8-afac-dcac8cca5f89']

### ISSUE THREAD

**Issue by jcald1** [postmanlabs/newman#2217]
I'm running Newman as a library.  I executed the following command
```
newman.run({
  environment: '7476980-b2dda329-72ea-40b8-afac-dcac8cca5f89',
  timeout: 1000000,
  timeoutRequest: 1000000,
  timeoutScript: 1000000,
  bail: true,
  reporters: 'cli,json',
  collection: 'https://api.getpostman.com/collections/7337586-64084678-a735-4b2e-9292-163bccfe6eca?apikey=<MY_API_KEY>'
}...
```

and got the following error:

```
[Error: ENOENT: no such file or directory, open '7476980-b2dda329-72ea-40b8-afac-dcac8cca5f89'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '7476980-b2dda329-72ea-40b8-afac-dcac8cca5f89',
  help: 'unable to read data from file "7476980-b2dda329-72ea-40b8-afac-dcac8cca5f89"'
}
```

It's not clear from the documentation whether running Newman as a library works with URLs or only files.  The example just uses a file, but there's also a CLI example that uses a URL.

I received a similar error when attempting to pass the uid of an environment instead of a local file.

**Comment by jimc404** [comment]
Newman as a library **works** with **URLs as well as files**

You need to provide the ```postmanApiKey``` key in the object you are passing to ```newman.run()```

It should be something like
```
newman.run({
environment: '<environment id or path>',
postmanApiKey: '<your api key>', // add this to your code
collection: '<collection id or path>',
...
```

**Comment by jimc404** [comment]
@jcald1 if that's the API key to your account in the code you've pasted, please edit the issue and delete it, also change it in your account.

If the user doesn't come online can a maintainer do it @codenirvana @shamasis

**Comment by jimc404** [comment]
This also uncovers a minor bug, if the user wants to fetch the collection but does not provide the api key, the process should exit after giving an error for the same. 

I want to work on a fix for that, should I open a new issue for that?

**Comment by jcald1** [comment]
> @jcald1 if that's the API key to your account in the code you've pasted, please edit the issue and delete it, also change it in your account.
> 
> If the user doesn't come online can a maintainer edit the issue @codenirvana @shamasis

Took care of it.  Thanks

**Comment by jcald1** [comment]
@jimc404, https://github.com/postmanlabs/newman/pull/2218 for the documentation.

**Comment by DannyDainton** [comment]
Hey @jcald1 

I'm going to close this issue out for now as it's just waiting on the docs changes.

Thank you for submitting the PR 🏆🚀
